### PR TITLE
Update code generator to support file update.

### PR DESCRIPTION
**Pull request by acbeni**
Previously the image upload wasn't working properly (it was more or less a placeholder), this pr tries to fix this by handling this ignored use case.

### CODE REVIEW

**Comment by katmatt** [comment]
@acbeni Without more context it's hard to properly review this PR. Can you please add a short description to this PR that explains this change a bit more?

**Comment by acbeni** [comment]
@katmatt sorry my bad here 😓 , i fixed the description, i hope this is more telling 👀

File: languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/client/files_producers/ClientConstants.kt [hunk 1]
```diff
@@ -14,10 +15,14 @@ class ClientConstants @Inject constructor(
         val basePackageName: String
 ) {
 
-    val requestUtilsPackage = "${sharedPackage}/utils/requests-utils"
-    val uriUtilsPackage = "${sharedPackage}/utils/uri-utils"
-    val commonTypesPackage = "${sharedPackage}/utils/common-types"
-    val middlewarePackage = "${sharedPackage}/utils/middleware"
-    val apiRoot = "$clientPackage/api-root"
+    val requestUtilsPackage = "${sharedPackage.lowerCasePackage()}/utils/requests-utils"
```
Comment (katmatt) [inline languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/client/files_producers/ClientConstants.kt:18]: I don't think that this will break our code, but why was this change necessary? I thought that the user provides the `sharedPackage` when he invokes the generator? Then he could fix it in his invaocation? Or am I missing something here?

File: languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt [hunk 2]
```diff
@@ -191,15 +193,20 @@ class JavaHttpRequestRenderer @Inject constructor(override val vrapTypeProvider:
             |   httpRequestPath += "?" + String.join("&", params);
             |}
         """.trimMargin().escapeAll()
-        
+        val bodySetter: String = if(bodyName != null){
+            if(this.bodies[0].type.isFile())
+                "try{httpRequest.setBody(Files.readAllBytes(file.toPath()));}catch(Exception e){e.printStackTrace();}"
```
Comment (katmatt) [inline languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt:198]: I see 👍 But wouldn't it be better to use a logger here?

Comment (katmatt) [inline languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt:198]: And the lines:
```java
httpRequest.setBody(Files.readAllBytes(file.toPath()));
```
load the file into memory first, ideally we would stream the contents here. But may be it's fine for now 😉

File: languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/client/files_producers/ClientConstants.kt (hunk 1 above)
Comment (acbeni) [reply languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/client/files_producers/ClientConstants.kt:18]: actually there are packages that don't pass through the `VrapType` provider. In those cases you end up with two package naming, one lower case and one not. this is why i decided to unify it here ⬆️

File: languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt (hunk 2 above)
Comment (acbeni) [reply languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt:198]: Nice catch, will see this 👍

Comment (acbeni) [reply languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt:198]: Yeah  using multipart upload might be the best way, but i just did it like this for sake of simplicity. we might come back to this.

Comment (acbeni) [reply languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt:198]: > Yeah using multipart upload might be the best way, but i just did it like this for sake of simplicity. we might come back to this if it causes trouble.

Comment (acbeni) [reply languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt:198]: Found out the reason, it was actually that we dont have a logger abstraction/convention. maybe i can leave it as here and create a separate issue for logging @katmatt ?

**Comment by katmatt** [comment]
Thanks for the explanation and they make a lot of sense to me 👍
